core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,322 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controllers, services and filters for solution.
+ */
+
+oppia.controller('StateSolution', [
+  '$scope', '$rootScope', '$modal', '$filter', '$injector',
+  'editorContextService', 'alertsService', 'INTERACTION_SPECS',
+  'stateSolutionService', 'explorationStatesService',
+  'oppiaExplorationHtmlFormatterService', 'stateInteractionIdService',
+  'stateHintsService', 'UrlInterpolationService', 'SolutionObjectFactory',
+  'responsesService', 'AnswerClassificationService',
+  'explorationContextService', 'angularNameService', 'oppiaHtmlEscaper',
+  function(
+    $scope, $rootScope, $modal, $filter, $injector,
+    editorContextService, alertsService, INTERACTION_SPECS,
+    stateSolutionService, explorationStatesService,
+    oppiaExplorationHtmlFormatterService, stateInteractionIdService,
+    stateHintsService, UrlInterpolationService, SolutionObjectFactory,
+    responsesService, AnswerClassificationService,
+    explorationContextService, angularNameService, oppiaHtmlEscaper) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.isActive = false;
+    $scope.answerIsExclusive = false;
+    $scope.answerCheckPassed = false;
+    $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+    $rootScope.interactionHtml = '';
```

Comment:
Please don't declare things on rootScope. That's very much an antipattern (and we want to remove the few instances of this from the other files). Ditto below.